### DISCUSSION TITLE
Make sure time-daemon is not available on the system or remove it during provisioning

### DISCUSSION THREAD

**Discussion by nckrtl** [serversideup/spin#117] [Ideas]
When provisioning a fresh Hetzner amd vps with Spin I always have to first remove `time-daemon` before provisioning. Else I get this error:
```
TASK [serversideup.spin.linux_common : Install latest versions of common packages.] *******************************************************************************
fatal: [<my-vps-ip>]: FAILED! => {"cache_update_time": 1728502981, "cache_updated": true, "changed": false, "msg": "'/usr/bin/apt-get -y -o \"Dpkg::Options::=--force-confdef\" -o \"Dpkg::Options::=--force-confold\"       install 'curl=8.5.0-2ubuntu10.4' 'figlet=2.2.5-3' 'fail2ban=1.0.2-3ubuntu0.1' 'mailutils=1:3.17-1.1build3' 'ncdu=1.19-0.1' 'ntp=1:4.2.8p15+dfsg-2~1.2.2+dfsg1-4build2' 'python3-minimal=3.12.3-0ubuntu2' 'python3-pip=24.0+dfsg-1ubuntu1' 'ssh=1:9.6p1-3ubuntu13.5' 'unzip=6.0-28ubuntu4.1' 'zip=3.0-13build1'' failed: E: Error, pkgProblemResolver::Resolve generated breaks, this may be caused by held packages.\n", "rc": 100, "stderr": "E: Error, pkgProblemResolver::Resolve generated breaks, this may be caused by held packages.\n", "stderr_lines": ["E: Error, pkgProblemResolver::Resolve generated breaks, this may be caused by held packages."], "stdout": "Reading package lists...\nBuilding dependency tree...\nReading state information...\nSome packages could not be installed. This may mean that you have\nrequested an impossible situation or if you are using the unstable\ndistribution that some required packages have not yet been created\nor been moved out of Incoming.\nThe following information may help to resolve the situation:\n\nThe following packages have unmet dependencies:\n ntpsec : Conflicts: time-daemon\n systemd-timesyncd : Conflicts: time-daemon\n", "stdout_lines": ["Reading package lists...", "Building dependency tree...", "Reading state information...", "Some packages could not be installed. This may mean that you have", "requested an impossible situation or if you are using the unstable", "distribution that some required packages have not yet been created", "or been moved out of Incoming.", "The following information may help to resolve the situation:", "", "The following packages have unmet dependencies:", " ntpsec : Conflicts: time-daemon", " systemd-timesyncd : Conflicts: time-daemon"]}
```

To remove time-daemon I run `sudo apt-get remove time-daemon` perhaps this could be added to the provisioning script? If there are other solutions I'm all ears.

**Comment by nckrtl** [comment]
ah nvm about this one, you cant remove time-daemon. I must've fixed the error differently. Not sure yet, closing for now

**Reply by jaydrogers** [reply]
I had issues with Hetzner today too.

The server was reporting the cache was updated, but the default Hetzner server image didn't have it's cache updated.

So I released a new Ansible Collection version: https://github.com/serversideup/ansible-collection-spin/releases/tag/v1.3.0

This release will force the cache to get updated on provision.

Not sure if this is related, but wanted to make you aware.

To get this, you should see `serversideup/ansible-collection-spin v1.3.0` somewhere at the beginning. If not, run `spin prune` and it will clear the collection cache and download it the next time you run `spin provision`.

**Reply by nckrtl** [reply]
Did a spin prune but with a fresh install I was still getting the error. Now fixed it by running `sudo apt-get update
sudo apt-get upgrade -y` on the fresh vps, and then provision. Maybe thats the same result as the updated cache, and for some reason im not using the latest ansible collection?

**Reply by jaydrogers** [reply]
That’s the same behavior I was getting too. It just started today for some reason.

Not sure why you wouldn’t get the latest version of the collection. You should see it download and see the version number in the logs after you run a `spin prune` and a `spin provision` again

**Reply by nckrtl** [reply]
Yeah indeed, will have to continue tomorrow evening. Got to hit the bed. Thanks a lot for all the help as always!
